data/registry_model/test/test_registry_proxy_model.py

```diff
@@ -1213,3 +1213,107 @@ def test_doesnt_bump_tag_expiration_when_upstream_is_dead(
             tag = proxy_model.get_repo_tag(repo_ref, self.tag)
         assert tag is not None
         assert tag.lifetime_end_ms == first_tag.lifetime_end_ms
+
+
+class TestPruningLRUProxiedImagesToAllowBlobUpload:
+    upstream_registry = "docker.io"
+    upstream_repository = "library/busybox"
+    orgname = "proxy-cache"
+    repository = f"{orgname}/{upstream_repository}"
+    tag = "1.35.0"
+
+    @pytest.fixture(autouse=True)
+    def setup(self, app):
+        self.user = get_user("devtable")
+        self.org = create_organization(self.orgname, "{self.orgname}@devtable.com", self.user)
+        self.org.save()
+        self.config = create_proxy_cache_config(
+            org_name=self.orgname,
+            upstream_registry=self.upstream_registry,
+            expiration_s=3600,
+        )
+
+    @patch("data.registry_model.registry_proxy_model.Proxy", MagicMock())
+    def test_auto_pruning_skipped_if_no_quota_set(self, create_repo):
+        repo_ref = create_repo(self.orgname, self.upstream_repository, self.user)
+        input_manifest = parse_manifest_from_bytes(
+            Bytes.for_string_or_unicode(UBI8_8_4_MANIFEST_SCHEMA2),
+            DOCKER_SCHEMA2_MANIFEST_CONTENT_TYPE,
+        )
+        proxy_model = ProxyModel(
+            self.orgname,
+            self.upstream_repository,
+            self.user,
+        )
+        assert proxy_model._check_blob_upload_and_prune(repo_ref, input_manifest) is None
+
+    @patch("data.registry_model.registry_proxy_model.Proxy", MagicMock())
+    def test_auto_pruning_skipped_for_manifest_list(self, create_repo, initialized_db):
+        repo_ref = create_repo(self.orgname, self.upstream_repository, self.user)
+        input_manifest = parse_manifest_from_bytes(
+            Bytes.for_string_or_unicode(testdata.PYTHON_LATEST["manifest"]),
+            DOCKER_SCHEMA2_MANIFESTLIST_CONTENT_TYPE,
+        )
+        proxy_model = ProxyModel(
+            self.orgname,
+            self.upstream_repository,
+            self.user,
+        )
+        limit_bytes = 2048
```

Comment:
let's set this to 0? this way we ensure that pulling a manifest list requires no space free in the namespace.

Reply:
removed setting quota size all together since the `upstream_manifest.is_manifest_list` moved to the very beginning of the func. I added an `assert` to verify ns_size doesn't change when a `manifest_list` is passed

Reply:
Good thinking!